src/containerapp-compose/azext_containerapp_compose/vendored_sdks/azext_containerapp/commands.py

```diff
@@ -0,0 +1,128 @@
+    except:
+        result['fqdn'] = None
+
+    return result
+
+
+def transform_containerapp_list_output(apps):
+    return [transform_containerapp_output(a) for a in apps]
+
+
+def transform_revision_output(rev):
+    props = ['name', 'active', 'createdTime', 'trafficWeight', 'healthState', 'provisioningState', 'replicas']
+    result = {k: rev['properties'][k] for k in rev['properties'] if k in props}
+
+    if 'name' in rev:
+        result['name'] = rev['name']
+
+    if 'fqdn' in rev['properties']['template']:
+        result['fqdn'] = rev['properties']['template']['fqdn']
+
+    return result
+
+
+def transform_revision_list_output(revs):
+    return [transform_revision_output(r) for r in revs]
+
+
+def load_command_table(self, _):
+    with self.command_group('containerapp', is_preview=True) as g:
+        g.custom_show_command('show', 'show_containerapp', table_transformer=transform_containerapp_output)
+        g.custom_command('list', 'list_containerapp', table_transformer=transform_containerapp_list_output)
+        g.custom_command('create', 'create_containerapp', supports_no_wait=True, exception_handler=ex_handler_factory(), table_transformer=transform_containerapp_output)
+        g.custom_command('update', 'update_containerapp', supports_no_wait=True, exception_handler=ex_handler_factory(), table_transformer=transform_containerapp_output)
+        g.custom_command('delete', 'delete_containerapp', supports_no_wait=True, confirmation=True, exception_handler=ex_handler_factory())
+        g.custom_command('exec', 'containerapp_ssh', validator=validate_ssh)
+        g.custom_command('up', 'containerapp_up', supports_no_wait=False, exception_handler=ex_handler_factory())
+        g.custom_command('browse', 'open_containerapp_in_browser')
+
+    with self.command_group('containerapp replica', is_preview=True) as g:
+        g.custom_show_command('show', 'get_replica')  # TODO implement the table transformer
+        g.custom_command('list', 'list_replicas')
+
+    with self.command_group('containerapp logs', is_preview=True) as g:
+        g.custom_show_command('show', 'stream_containerapp_logs', validator=validate_ssh)
+
+    with self.command_group('containerapp env') as g:
+        g.custom_show_command('show', 'show_managed_environment')
+        g.custom_command('list', 'list_managed_environments')
+        g.custom_command('create', 'create_managed_environment', supports_no_wait=True, exception_handler=ex_handler_factory())
+        g.custom_command('delete', 'delete_managed_environment', supports_no_wait=True, confirmation=True, exception_handler=ex_handler_factory())
+
+    with self.command_group('containerapp env dapr-component') as g:
+        g.custom_command('list', 'list_dapr_components')
+        g.custom_show_command('show', 'show_dapr_component')
+        g.custom_command('set', 'create_or_update_dapr_component')
+        g.custom_command('remove', 'remove_dapr_component')
```

Comment:
Why do you vendor the source code of other extension into `vendored_sdks`?
We usually don't recommend this, because it may introduce some extra problems: such as there may be conflicts between installing extension and vendored extension, and increase the maintenance costs to keep extension source code and vendored extension source code consistent (especially for bug fix)

Reply:
I vendored it because the guidance I read was that there wasn't a way to require dependencies between extensions. 

I do check to see if an installed version is available and only load the vendored version if not.

I've also got a job ready to run to continue to track the containerapp extension and make sure use cases don't break.

Eventually the feature in this extension will likely be migrated to the containerapp extension, once we get some usage and feedback.

Reply:
@smurawski We usually suggest that the commands of this extension be migrated to `containerapp` extension in this case, because there are dependencies between those two extensions and `containerapp-compose` actually belongs to `containerapp`.
Therefore, we do not recommend splitting the command groups with high coupling into two different extensions to increase the complexity and maintenance cost of system.